source/Halibut.Tests/Builders/PendingRequestQueueBuilder.cs

```diff
@@ -9,6 +9,7 @@ public class PendingRequestQueueBuilder
         ILog? log;
         string? endpoint;
         TimeSpan? pollingQueueWaitTimeout;
+        bool? relyOnConnectionTimeoutsInsteadOfPollingRequestMaximumMessageProcessingTimeout;
```

Comment:
Could we have this as non nullable, and just default it to false?

Reply:
I personally would prefer to use the default from `HalibutTimeoutsAndLimits` by default, and only use this if the test has actually chosen to change it.

Reply:
Oh i see the complexity lower down, probably is better here :+1: